client/rest-high-level/src/main/java/org/elasticsearch/client/security/CreateApiKeyResponse.java

```diff
@@ -79,13 +85,22 @@ public boolean equals(Object obj) {
     }
 
     static final ConstructingObjectParser<CreateApiKeyResponse, Void> PARSER = new ConstructingObjectParser<>("create_api_key_response",
-            args -> new CreateApiKeyResponse((String) args[0], (String) args[1], new SecureString((String) args[2]),
-                    (args[3] == null) ? null : Instant.ofEpochMilli((Long) args[3])));
+        args -> {
+            final String id = (String) args[1];
+            final String key = (String) args[2];
+            if (args[4] != null
+                && false == args[4].equals(Base64.getEncoder().encodeToString((id + ":" + key).getBytes(StandardCharsets.UTF_8)))) {
```

Comment:
Would it make sense to have a
```
private static String encode(CharSequence id, CharSequence key) {
    return Base64.getEncoder().encodeToString((id + ":" + key).getBytes(StandardCharsets.UTF_8));
} 
```
and then call it from here and in `getEncoded()` ?

Reply:
Yes, good idea.